flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/utils/StatusRecorderTest.java

```diff
@@ -70,49 +74,64 @@ public void testPatchOnlyWhenChanged() throws InterruptedException {
 
     @Test
     public void testFlinkDeploymentMetrics() throws InterruptedException {
-        var metricListener = new MetricListener();
+        var metrics = new HashMap<String, Metric>();
```

Comment:
This is boilerplate, do not understand why you prefer this approach over the `MetricListener`

Reply:
It's not very easy to use for these simple tests due to some hardcoded/not exposed logic. I added a more details in the other comment :)